apps/users/models.py

```diff
@@ -361,10 +351,19 @@ def source(self):
 
     @property
     def name(self):
-        return smart_unicode(self.display_name or self.username)
+        if self.display_name:
+            return smart_unicode(self.display_name)
+        elif self.has_anonymous_username:
+            return _('Anonymous')
+        else:
+            return smart_unicode(self.username)
 
     welcome_name = name
 
+    @property
```

Comment:
Does this have to be a property? I don't really like making things properties unless it's needed to satisfy some kind of external interface constraint. Properties conceal the fact that you're executing code without adding any benefits.